File: src/lib/api/permissions.ts
import api from './client';

export interface Permission {
  key: string;
  label: string;
  description: string;
  category: string;
}

export interface UserPermissions {
  id: string;
  email?: string;
  full_name: string;
  role: string;
  permissions: string[];
  updated_at: string;
}

export const permissionsApi = {
  getAvailable: async (): Promise<Permission[]> => {
    const response = await api.get('/permissions/available');
    return response.data;
  },

  getUserPermissions: async (userId: string): Promise<UserPermissions> => {
    const response = await api.get(`/permissions/user/${userId}`);
    return response.data;
  },

  updateUserPermissions: async (userId: string, permissions: string[]): Promise<UserPermissions> => {
    const response = await api.put(`/permissions/user/${userId}`, { user_id: userId, permissions });
    return response.data;
  },

  getAllStaffPermissions: async (): Promise<UserPermissions[]> => {
    const response = await api.get('/permissions/staff');
    return response.data;
  },
};

File: backend/app/schemas/permissions.py
"""Permission Management Schemas"""
from pydantic import BaseModel
from typing import List, Optional


class UserPermissionsUpdate(BaseModel):
    """Update user permissions"""
    user_id: str
    permissions: List[str]


class UserPermissionsResponse(BaseModel):
    """User with permissions"""
    id: str
    email: Optional[str]
    full_name: str
    role: str
    permissions: List[str] = []
    assigned_location_id: Optional[str] = None
    updated_at: Optional[str] = None

    class Config:
        from_attributes = True


class AvailablePermission(BaseModel):
    """Available permission definition"""
    key: str
    label: str
    description: str
    category: str
